lib/crypt-md5.c

```diff
@@ -72,23 +74,39 @@ crypt_md5crypt_rn (const char *phrase, size_t phr_size,
   MD5_CTX *ctx = &buf->ctx;
   uint8_t *result = buf->result;
   char *cp = (char *)output;
-  const char *salt = setting;
+  char *salt = (char *)buf->setting_tmp;
 
   size_t salt_size;
   size_t cnt;
 
+  /* Copy the setting to a temporary buffer. */
+  XCRYPT_STRCPY_OR_ABORT (salt, CRYPT_OUTPUT_SIZE - 1, setting);
```

Comment:
I think it should be possible to implement these changes (and ditto in sha*crypt) without salt string copying and with less code.

Reply:
Fixed in rebased commits.